src/status_im/common/validation/profile.cljs

```diff
@@ -0,0 +1,42 @@
+(ns status-im.common.validation.profile
+  (:require [clojure.string :as string]
+            [utils.i18n :as i18n]))
+
+;; NOTE - validation should match with Desktop
+;; https://github.com/status-im/status-desktop/blob/2ba96803168461088346bf5030df750cb226df4c/ui/imports/utils/Constants.qml#L468
+(def min-length 5)
+
+(def max-length 24)
+
+(def emoji-regex
+  #"(\u00a9|\u00ae|[\u2000-\u3300]|\ud83c[\ud000-\udfff]|\ud83d[\ud000-\udfff]|\ud83e[\ud000-\udfff])")
+
+(defn has-emojis [s] (boolean (re-find emoji-regex s)))
+
+(def common-names ["Ethereum" "Bitcoin"])
+
+(defn has-common-names [s] (pos? (count (filter #(string/includes? s %) common-names))))
+
+(def status-regex #"^[a-zA-Z0-9\-_ ]+$")
+
+(defn has-special-characters [s] (not (re-find status-regex s)))
+
+(defn name-too-short [s] (< (count (string/trim (str s))) min-length))
+
+(defn name-too-long [s] (> (count (string/trim (str s))) max-length))
```

Comment:
@mohsen-ghafouri, I forgot to mention predicates in Clojure should end with question mark. Reference CSG https://guide.clojure.style/#naming-predicates